src/main/kotlin/no/fdk/dataset_catalog/extensions/DatasetExtensions.kt

```diff
@@ -1,5 +1,258 @@
+        temporal = temporal?.map {
+            PeriodOfTimeDBO(
+                startDate = it.startDate,
+                endDate = it.endDate
+            )
+        },
+        spatial = spatial?.mapNotNull { it.uri },
+        accessRight = accessRights?.uri,
+        legalBasisForRestriction = legalBasisForRestriction?.map { UriWithLabel(it.uri, it.prefLabel) },
+        legalBasisForProcessing = legalBasisForProcessing?.map { UriWithLabel(it.uri, it.prefLabel) },
+        legalBasisForAccess = legalBasisForAccess?.map { UriWithLabel(it.uri, it.prefLabel) },
+        accuracy = hasAccuracyAnnotation?.let {
+            QualityAnnotationDBO(
+                inDimension = it.inDimension,
+                motivatedBy = it.motivatedBy,
+                hasBody = it.hasBody
+            )
+        },
+        completeness = hasCompletenessAnnotation?.let {
+            QualityAnnotationDBO(
+                inDimension = it.inDimension,
+                motivatedBy = it.motivatedBy,
+                hasBody = it.hasBody
+            )
+        },
+        currentness = hasCurrentnessAnnotation?.let {
+            QualityAnnotationDBO(
+                inDimension = it.inDimension,
+                motivatedBy = it.motivatedBy,
+                hasBody = it.hasBody
+            )
+        },
+        availability = hasAvailabilityAnnotation?.let {
+            QualityAnnotationDBO(
+                inDimension = it.inDimension,
+                motivatedBy = it.motivatedBy,
+                hasBody = it.hasBody
+            )
+        },
+        relevance = hasRelevanceAnnotation?.let {
+            QualityAnnotationDBO(
+                inDimension = it.inDimension,
+                motivatedBy = it.motivatedBy,
+                hasBody = it.hasBody
+            )
+        },
+        references = references?.map {
+            ReferenceDBO(
+                referenceType = it.referenceType?.code,
+                source = it.source?.uri
+            )
+        },
+        relatedResources = relations?.map { UriWithLabel(it.uri, it.prefLabel) },
+        provenance = provenance?.uri,
+        frequency = accrualPeriodicity?.uri,
+        conformsTo = conformsTo?.map { UriWithLabel(it.uri, it.prefLabel) },
+        informationModelsFromOtherSources = informationModel?.map { UriWithLabel(it.uri, it.prefLabel) },
+        informationModelsFromFDK = informationModelsFromFDK,
+        qualifiedAttributions = qualifiedAttributions,
+        type = type,
+        inSeries = inSeries,
+        seriesDatasetOrder = seriesDatasetOrder
+    )
+
+
+fun DatasetDBO.toDataset(): Dataset {
+    return Dataset(
+        id = id,
+        catalogId = catalogId,
+        uri = uri,
+
+        specializedType = specializedType,
+        originalUri = originalUri,
+        lastModified = lastModified,
+        registrationStatus = when {
+            published == true -> REGISTRATION_STATUS.PUBLISH
+            approved == true -> REGISTRATION_STATUS.APPROVE
```

Comment:
trenger ikke `== true` på en boolean
```suggestion
            published -> REGISTRATION_STATUS.PUBLISH
            approved -> REGISTRATION_STATUS.APPROVE
```